framework/src/controller/application.js

```diff
@@ -86,36 +87,49 @@ class Application {
 		constants = {},
 		config = { components: { logger: null }, modules: {} }
 	) {
-		if (typeof label === 'function') {
-			label = label.call();
+		let appConfig;
+
+		// If user passes multiple config objects merge them in left-right order
+		if (Array.isArray(config)) {
```

Comment:
Constructor config is initializing as object and treating as array is bit strange.
Please use consistent type

Reply:
It was intended to pass multiple custom config files. Remember for application network config or custom config are both custom configuration. 

This particular part will be removed with the PR of https://github.com/LiskHQ/lisk/issues/3171 where we have config compilation logic separated and application will accept unified compile configuration.